### PR TITLE
Add support for GHC 8.8

### PR DESCRIPTION
**Pull request by akshaymankar**
Fixes https://github.com/bitemyapp/bloodhound/issues/273
This will also require a version bump to quickcheck-arbitrary-template, I created a PR for that here: https://github.com/bitemyapp/quickcheck-arbitrary-template/pull/1. Once that is merged I will update the URL in stack-8.8.yaml and make this PR reviewable.

### CODE REVIEW

**Comment by wraithm** [comment]
@akshaymankar There's also https://github.com/plow-technologies/quickcheck-arbitrary-template/pull/9 It'd be great if that got merged and pushed up to hackage first.

**Comment by akshaymankar** [comment]
Given the silence on plow-technologies/quickcheck-arbitrary-template#9, I think it is better to keep bitemyapp/quickcheck-arbitrary-template as the dependency. Maybe it can be changed in a separate PR when the maintainers from plow-technologies respond.

**Comment by bitemyapp** [comment]
Looks like the `UpdatableIndexSetting` isomorphism-ish is on the fritz again. Obnoxious.

**Comment by akshaymankar** [comment]
> Looks like the `UpdatableIndexSetting` isomorphism-ish is on the fritz again. Obnoxious.

The problem is with [`CharFilterDefinitionMapping`](https://github.com/bitemyapp/bloodhound/blob/c5603ff4e120b7b23960373541d8ecfe142f09f3/src/Database/Bloodhound/Internal/Analysis.hs#L63). the test fails whenever the generated `CharFilterDefinition` is constructed with `CharFilterDefinitionMapping` and one of the mappings contains `=>`.  
I think it would be best to represent it as `Map Text (Text, Text)` instead of `Map Text Text`. This would still require a custom instance of `Arbitrary`, as we don't have any way of escaping `=>` (at least it is not documented [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-mapping-charfilter.html)).

**Comment by akshaymankar** [comment]
@bitemyapp Is it ok if that flakey test is resolved separately?

**Comment by Daniel-Diaz** [comment]
PR https://github.com/plow-technologies/quickcheck-arbitrary-template/pull/9 is now merged and a new version was uploaded to Hackage: http://hackage.haskell.org/package/quickcheck-arbitrary-template-0.2.1.1.

File: src/Database/Bloodhound/Internal/Query.hs [hunk 1]
```diff
@@ -1620,7 +1620,7 @@ functionScoreFunctionsPair (FunctionScoreSingle fn)
 functionScoreFunctionsPair (FunctionScoreMultiple componentFns) =
   ("functions", toJSON componentFns)
 
-fieldTagged :: Monad m => (FieldName -> Object -> m a) -> Object -> m a
+fieldTagged :: (Monad m, MonadFail m)=> (FieldName -> Object -> m a) -> Object -> m a
```
Comment (wraithm) [inline src/Database/Bloodhound/Internal/Query.hs:1623]: I tried building this with stack-8.6.yaml, and it failed. I think you'd need to import `MonadFail` for that compiler version.

Comment (wraithm) [reply src/Database/Bloodhound/Internal/Query.hs:1623]:
```haskell
bloodhound          >     Not in scope: type constructor or class ‘MonadFail’
bloodhound          >     Perhaps you meant ‘MonadFix’ (imported from Bloodhound.Import)
bloodhound          >      |
bloodhound          > 1623 | fieldTagged :: (Monad m, MonadFail m)=> (FieldName -> Object -> m a) -> Object -> m a
bloodhound          >      |```

Comment (akshaymankar) [reply src/Database/Bloodhound/Internal/Query.hs:1624]: Thanks for finding this. I also went through older versions and adjusted imports accordingly. I also bumped LTS versions for them, only GHC 8.0 had a major LTS version bumped, it compiles, so hopefully, it shouldn't be a problem.

**Review by wraithm — APPROVED** [review]

**Pull request merged**